diff --git a/web/pub/js/inGraph/ext-inGraph/Ext.iG.FlotPanel.js b/web/pub/js/inGraph/ext-inGraph/Ext.iG.FlotPanel.js
index 341c3a3..1f6d453 100644
--- a/web/pub/js/inGraph/ext-inGraph/Ext.iG.FlotPanel.js
+++ b/web/pub/js/inGraph/ext-inGraph/Ext.iG.FlotPanel.js
@@ -123,7 +123,7 @@ Ext.iG.FlotPanel = Ext.extend(Ext.Panel, {
                                 });
                             });
                             
-                            if(!this.templateWindow) {                   	    
+                            if(!this.templateWindow) {            	    
 	                            this.templateWindow = new Ext.iG.TemplateWindow({
 	                            	store : this.store,
 	                            	listeners : {
@@ -306,18 +306,32 @@ Ext.iG.FlotPanel = Ext.extend(Ext.Panel, {
         });
         
         this.add({
+            xtype : 'box',
+            id : this.id + '-labels',
+            height : 20
+        });
+        
+        this.insert(0, {
         	store	: this.store,
+        	flotOptions : {
+        		legend : {
+        			container : '#' + this.id + '-labels',
+        			noColumns : 5
+        		}
+        	},
         	ref		: 'flot'
         });
         
         if(this.overview) {
-        	this.add([{
-        		xtype   : 'component',
-        		autoEl  : {
-        			tag : 'hr',
-        			cls : 'iG-hs'
-        		}
-        	}, {
+            this.insert(1, {
+                xtype   : 'component',
+                autoEl  : {
+                    tag : 'hr',
+                    cls : 'iG-hs'
+                }
+            });
+
+        	this.insert(2, {
         		autoAddYAxes : false,
         	    flotOptions : {
         	        xaxis       : {
@@ -365,7 +379,7 @@ Ext.iG.FlotPanel = Ext.extend(Ext.Panel, {
         	    	}, this.store.baseParams)
         	    }),
         	    ref			: 'overview'
-        	}]);
+        	});
         	
         	this.flot.on({
         		zoomin	: function(flot, ranges) {
